docs/proposals/custom-sidecar-config.md

```diff
@@ -0,0 +1,32 @@
+# Custom Container Configuration
+
+## Introduction
+Users and field team have requested access to custom configuration of kubernetes sidecar and init containers. See [#3925](https://github.com/kumahq/kuma/issues/3925). This intuitively aligns with the ProxyTemplate configuration, which provides a free-form configuration of Envoy. The current plan is to introduce a new CRD, similar to ProxyTemplate, which can accommodate sidecar config.
+
+## Custom Resource Definition
+The new CRD will be named ContainerTemplate. It will allow for customer configuration of both sidecar and init containers.
+
+This will follow a similar format to ProxyTemplate. A ContainerTemplate will match pods via namespaces and labels, and modify the containers associated with those pods. The spec will contain full templates for “sidecar” and “init” dataplane containers.
+
+### Example
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: ContainerTemplate
+metadata:
+  name: container-template-1
+spec:
+  selectors:
+    - namespace: namespace-1
+  sidecarTemplate: | …
+  initTemplate: | …
```

Comment:
Would be useful to have an actual real life example (.e.g disable priviledged containers). Without I struggle to understand how this would actually work.